app/models/ems_refresh/parsers/azure.rb

```diff
@@ -0,0 +1,214 @@
+        return uid, new_result
+      end
+
+      def fetch_az(availability_zone)
+        @data_index.fetch_path(:availability_zones, availability_zone.downcase)
+      end
+
+      def process_os(vm)
+        {
+          :product_name => guest_os(vm)
+        }
+      end
+
+      def guest_os(vm)
+        image_reference = vm.fetch_path('properties', 'storageProfile', 'imageReference')
+        image_reference['offer'] + " " + image_reference['sku'].gsub('-', ' ')
+      end
+
+      def populate_hardware_hash_with_disks(hardware_disks_array, vm)
+        data_disks = vm.fetch_path('properties', 'storageProfile', 'dataDisks')
+        data_disks.each do |disk|
+          disk_size      = disk['diskSizeGB'] * 1.gigabyte
+          disk_name      = disk['name']
+          disk_location  = disk['vhd']['uri']
+
+          add_instance_disk(hardware_disks_array, disk_size, disk_name, disk_location)
+        end
+      end
+
+      def add_instance_disk(disks, size, name, location)
+        super(disks, size, name, location, "azure")
+      end
+
+      def populate_hardware_hash_with_networks(hardware_networks_array, vm)
+        nics = vm.fetch_path('properties', 'networkProfile', 'networkInterfaces')
+
+        nics.each do |n|
+          n['properties'].each do |n_prop|
+            private_network = {
+              :ipaddress => n_prop['properties']['privateIPAddress'],
+              :hostname  => n_prop['name']
+            }.delete_nils
+
+            public_network = {
+              :ipaddress => n_prop['properties']['publicIPAddress'],
+              :hostname  => n_prop['name']
+            }.delete_nils
+
+            hardware_networks_array <<
+              private_network.merge(:description => "private") unless private_network.blank?
+            hardware_networks_array <<
+              public_network.merge(:description => "public")   unless public_network.blank?
+          end
+          hardware_networks_array.flatten!
+        end
+      end
+
+      def populate_hardware_hash_with_series(hardware_hash, vm, series)
+        return if series.nil?
+        hardware_hash[:numvcpus]      = series[:cpus]
+        hardware_hash[:logical_cpus]  = series[:cpus]
+        hardware_hash[:memory_cpu]    = series[:memory] # MB
+        hardware_hash[:disk_capacity] = series[:os_disk_size] + series[:swap_disk_size]
+
+        os_disk = vm.fetch_path('properties', 'storageProfile', 'osDisk')
+        sz      = series[:os_disk_size]
+
+        add_instance_disk(hardware_hash[:disks], sz,  os_disk['name'], os_disk['vhd']) unless sz.zero?
+
+        # No data availbale on swap disk? Called temp or resource disk.
+      end
+
+      def clean_up_extra_flavor_keys
+        @data[:flavors].each do |f|
+          f.delete(:os_disk_size)
+          f.delete(:swap_disk_size)
+        end
+      end
+
+      def get_locations
+        @vmm.locations.collect do |location|
+          location.delete!(' ')
+          location.match(VALID_LOCATION).to_s
```

Comment:
This feels really odd to me, especially inside of a collect block.  It's basically saying, match the location and return the matched portion.  But what if there's no match?  Don't you get an empty string back?  Is that expected?